api/checks/api.go

```diff
@@ -44,31 +42,27 @@ type API interface {
 }
 
 type checksAPIImpl struct {
-	ctx   context.Context
+	shared.AppEngineAPI
+
 	queue string
 }
 
 // NewAPI returns a real implementation of the API
 func NewAPI(ctx context.Context) API {
-	return checksAPIImpl{
-		ctx:   ctx,
-		queue: CheckProcessingQueue,
+	return &checksAPIImpl{
```

Comment:
why is `&` added here?

Reply:
Without `&`, we are returning a struct, which does not satisfy the `API` interface (or more specifically, the `shared.AppEngineAPI` interface), as the embedded `shared.appEngineAPIImpl` has a few functions using the *pointer* receiver type (those functions need to modify the content of `appEngineAPIImpl`, e.g. for caching).

Reply:
But the `&` is used next to the `checksAPIImpl` and I don't see any `checksAPIImpl` functions use [pointer receiver type](https://github.com/web-platform-tests/wpt.fyi/blob/38ce5b2bae65f59f15d14283806fb2a4835e81d2/api/checks/api.go#L36)?

Reply:
Yeah actually you're right. I'm embedding the interface `AppEngineAPI` here, which is initialized to a pointer `& appEngineAPIImpl` created by `NewAppEngineAPI`, so there's our pointer required by the methods on `*appEngineAPIImpl`. `checksAPIImpl` itself doesn't have to be pointer.

```suggestion
	return checksAPIImpl{
```